frontend/src/modules/WellLogViewer/settings/components/TemplateTrackSettings/private-components/SortablePlotList.tsx

```diff
@@ -0,0 +1,223 @@
+                <AddItemButton buttonText="Add plot" options={PLOT_TYPE_OPTIONS} onOptionClicked={addPlot} />
+            </Label>
+
+            <SortableList onItemMoved={handleTrackMove}>
+                {props.plots.map((plot) => (
+                    <SortablePlotItem
+                        key={plot._id}
+                        plot={plot}
+                        curveHeaderOptions={curveHeaderOptions}
+                        onPlotUpdate={handlePlotUpdate}
+                        onDeletePlot={removePlot}
+                    />
+                ))}
+            </SortableList>
+        </div>
+    );
+}
+
+type SortablePlotItemProps = {
+    plot: TemplatePlotConfig;
+    curveHeaderOptions: CurveDropdownOption[];
+    onPlotUpdate: (plot: TemplatePlotConfig) => void;
+    onDeletePlot: (plot: TemplatePlotConfig) => void;
+};
+
+function SortablePlotItem(props: SortablePlotItemProps) {
+    const { onPlotUpdate } = props;
+    const secondCurveNeeded = props.plot.type === "differential";
+
+    function handlePlotChange(changes: Partial<TemplatePlotConfig>) {
+        const newPlot = makeTrackPlot({
+            ...props.plot,
+            ...changes,
+        });
+
+        onPlotUpdate(newPlot);
+    }
+
+    const title = (
+        <>
+            <Dropdown
+                placeholder="Select a curve"
+                value={props.plot._logAndName}
+                options={props.curveHeaderOptions}
+                onChange={(v) => handlePlotChange({ _logAndName: v, name: v.split("::")[1] })}
+            />
+        </>
+    );
+
+    const endAdornment = (
+        <>
+            {secondCurveNeeded && (
+                <>
+                    <DenseIconButton
+                        title="Swap curves"
+                        onClick={() =>
+                            handlePlotChange({
+                                _logAndName: props.plot._logAndName2,
+                                _logAndName2: props.plot._logAndName,
+                                name: props.plot.name2,
+                                name2: props.plot.name,
+                            })
+                        }
+                    >
+                        <SwapHoriz fontSize="inherit" />
+                    </DenseIconButton>
+
+                    <Dropdown
+                        placeholder="Select 2nd curve"
+                        value={props.plot._logAndName2}
+                        options={props.curveHeaderOptions}
+                        onChange={(v) => handlePlotChange({ _logAndName2: v, name2: v.split("::")[1] })}
+                    />
+                </>
+            )}
+            <div className="text-xs w-28 flex-shrink-0">
+                <Dropdown
+                    value={props.plot.type}
+                    options={PLOT_TYPE_OPTIONS}
+                    onChange={(v) => handlePlotChange({ type: v as TemplatePlotTypes })}
```

Comment:
`as TemplatePlotTypes` does not seem to be necessary.

Reply:
Yes. Also could get rid of the one used on "addPlot" aswell, just needed to update the `AddItemButton` comp to pass the type